framework/validators/CompareValidator.php

```diff
@@ -151,6 +151,17 @@ public function validateAttribute($model, $attribute)
             $compareAttribute = $this->compareAttribute === null ? $attribute . '_repeat' : $this->compareAttribute;
             $compareValue = $model->$compareAttribute;
             $compareLabel = $compareValueOrAttribute = $model->getAttributeLabel($compareAttribute);
+
+            if ($model->hasErrors($this->compareAttribute)) {
```

Comment:
why you check `$this->compareAttribute` instead `$compareAttribute`?

Reply:
it was accidentally, replaced